### PR TITLE
We use node 10 now

### LINKED ISSUE #7421: Node 10 is the current version now, let's update the docs and configs
In #7418, ops have upgraded the node version used in production. We should adjust the docs/configs.

### PR DESCRIPTION
**Pull request by willdurand**
Fixes #7421

---

This PR updates:

- the travis ci config
- our documentation
- the nvm/npm configs

### CODE REVIEW

**Comment by willdurand** [comment]
There are two issues with Node 11:

```
FAIL tests/unit/core/i18n/test_utils.js
  ● /home/travis/build/mozilla/addons-frontend/tests/unit/core/i18n/test_utils.js › makeI18n › falls-back to number.toLocaleString when Intl is missing a NumberFormat constructor
    expect(received).toEqual(expected)
    Expected value to equal:
      "12 345"
    Received:
      "12 345"
      489 |       const toLocaleStringSpy = sinon.spy(Number.prototype, 'toLocaleString');
      490 |       const number = 12345;
    > 491 |       expect(i18n.formatNumber(number)).toEqual('12 345');
          |                                         ^
      492 |       sinon.assert.calledWith(toLocaleStringSpy, 'fr');
      493 |       sinon.assert.notCalled(numberFormatSpy);
      494 |     });
      at Object.toEqual (tests/unit/core/i18n/test_utils.js:491:41)
FAIL tests/unit/amo/components/TestSearchResult.js
  ● /home/travis/build/mozilla/addons-frontend/tests/unit/amo/components/TestSearchResult.js › localises the user count
    Expected <span> to contain "5 253" but it did not.
    Expected HTML: "5 253"
    Actual HTML: "5 253 users"
      76 |     // `\xa0` is a non-breaking space.
      77 |     // See: https://github.com/airbnb/enzyme/issues/1349
    > 78 |     expect(root.find('.SearchResult-users-text')).toIncludeText('5\xa0253');
         |                                                   ^
      79 |   });
      80 | 
      81 |   it('renders the user count as singular', () => {
      at Object.toIncludeText (tests/unit/amo/components/TestSearchResult.js:78:51)
```

File: .travis.yml [hunk 1]
```diff
@@ -3,18 +3,18 @@ _aliases:
...
     # This is the NodeJS version we run in production.
     language: node_js
-    node_js: '8'
+    node_js: '10'
     before_install: npm install -g yarn
   - &node-next
     # This is the next NodeJS version we will support.
     language: node_js
-    node_js: '10'
+    node_js: '11'
```
Comment (rebmullin) [inline .travis.yml]: nit: shouldn't this be 12 (since you said we wont be using 11 anyway)?

Comment (willdurand) [reply .travis.yml]: 12 has not been released yet, iirc

Comment (rebmullin) [reply .travis.yml]: I guess I was going by this comment:

`This is the next NodeJS version we will support.` 

which you mentioned would be 12 above

Comment (willdurand) [reply .travis.yml]: Yep, because LTS versions are even numbers (6, 8, 10, etc.) and odd numbers are not.

**Pull request merged**